src/main/java/com/actiontech/dble/route/sequence/handler/DistributedSequenceHandler.java

```diff
@@ -202,8 +202,19 @@ public long nextId(String prefixName) throws SQLNonTransientException {
         if (time >= deadline) {
             throw new SQLNonTransientException("Global sequence has reach to max limit and can generate duplicate sequences.");
         }
-        if (threadLastTime.get() == null) {
-            threadLastTime.set(time);
+        Long lastTimestamp = threadLastTime.get();
+        if (lastTimestamp == null) {
+            if (time >= startTimeMilliseconds) {
+                threadLastTime.set(time);
+                lastTimestamp = time;
+            } else {
+                lastTimestamp = startTimeMilliseconds;
```

Comment:
lastTimestamp  init when the handler be loaded would be better

Reply:
keep it threadlocal